design-principles.md

```diff
@@ -0,0 +1,45 @@
+# Design Principles
+
+There are exceptions to these principles, but these are general guidelines the project strives to adhere to.
+
+## General
+
+- Bootkube should be a single-use tool, which only runs on the first node in a cluster.
+    - An exception is the `recovery` subcommand, discussed below.
+- Bootkube should not be required to run to add new nodes to a cluster.
+    - For example, adding nodes should not require a `bootkube join` command.
+- Should not require flag or configuration coordination between the `render` and `start` steps.
+    - Required flag coordination means certain `render` assets will only work with certain `start` flags, and this is something we should avoid.
+    - For example, `bootkube render --self-hosted-etcd` requires no changes when ultimately running `bootkube start`.
+- Avoid adding feature flags as much as possible. This makes testing & stability very difficult to maintain.
+    - Users should be able to modify assets after `render` and before `start`. This allows flexibility for any use-case without bootkube owning the complexity in the codebase.
```

Comment:
Maybe expand on this and make it more of an active voice? e.g. "Users customize their cluster by modifying the output of `bootkube render` to fit their configuration. Users can generate their own assets for use with `bootkube start`, subject to following a small number of conventions."

I think this is one of the strongest advantages of bootkube. I like the "complex rendering needs" part next, but as written right now it sounds a little scary.